src/webextension/survey/components/app.jsx

```diff
@@ -46,7 +46,16 @@ class App extends Component {
   }
 
   handleSubmit(values) {
-    sendMessage('submission', values);
+    Promise
+      .all([
+        browser.tabs.query({ active: true, currentWindow: true }),
+        browser.runtime.getBackgroundPage()
+      ])
+      .then(([ tabs, backgroundWindow ]) => {
+        const requests = backgroundWindow.app.httpObserver.get(tabs[0].id);
```

Comment:
Yeah, it's impossible to invoke this code without an active tab, since it happens in a pageAction popup, which requires a window, which always has an active tab.